spec/unit/govuk_index/publishing_event_processor_spec.rb

```diff
@@ -2,57 +2,47 @@
 require "govuk_message_queue_consumer/test_helpers"
 
 RSpec.describe GovukIndex::PublishingEventProcessor do
-  it_behaves_like "a message queue processor"
-
-  it "will process and ack a single message" do
-    message = double(
-      payload: {
-        "base_path" => "/cheese",
-        "document_type" => "help_page",
-        "title" => "We love cheese",
-      },
-      delivery_info: {
-        routing_key: "routing.key",
-      },
-    )
-
-    expect(GovukIndex::PublishingEventMessageProcessor).to receive(:process).with(
-      [["routing.key", message.payload]],
+  let(:content_item) { generate_random_example }
+  let(:message) do
+    GovukMessageQueueConsumer::MockMessage.new(
+      content_item,
+      {},
+      { routing_key: "routing.key" },
     )
+  end
+  let(:logger) { Logging.logger[described_class] }
 
-    expect(message).to receive(:ack)
+  it_behaves_like "a message queue processor"
 
-    subject.process(message)
+  context "when successful" do
+    before do
+      expect(GovukIndex::PublishingEventMessageProcessor)
+        .to receive(:process)
+        .with("routing.key", message.payload)
+    end
+
+    it "logs a message" do
+      expect(logger).to receive(:info).with(
+        /#{Regexp.escape("Processing message (attempt 1/5): {\"content_id\":\"#{content_item['content_id']}\"")}/,
+      )
+
+      described_class.new.process(message)
+    end
+
+    it "acknowledges the message" do
+      expect(message).to receive(:ack)
+      described_class.new.process(message)
+    end
   end
 
-  it "will process and ack an array of messages" do
-    message1 = double(
-      payload: {
-        "base_path" => "/cheese",
-        "document_type" => "help_page",
-        "title" => "We love cheese",
-      },
-      delivery_info: {
-        routing_key: "routing.key",
-      },
-    )
-    message2 = double(
-      payload: {
-        "base_path" => "/crackers",
-        "document_type" => "help_page",
-        "title" => "We love crackers",
-      },
-      delivery_info: {
-        routing_key: "routing.key",
-      },
-    )
-
-    expect(GovukIndex::PublishingEventMessageProcessor).to receive(:process).with(
-      [["routing.key", message1.payload], ["routing.key", message2.payload]],
-    )
-    expect(message1).to receive(:ack)
-    expect(message2).to receive(:ack)
+  context "when an error is raised" do
+    before do
+      expect(GovukIndex::PublishingEventMessageProcessor)
+        .to receive(:process)
+        .with("routing.key", message.payload)
+        .and_raise("oh no")
+    end
 
-    subject.process([message1, message2])
+    it_behaves_like "a retryable queue processor"
```

Comment:
Shall we move this to be situated next to `it_behaves_like "a message queue processor"` to indicate we intentionally want both?

Reply:
Oh I hadn't clocked that this was within a context - yeah leave as is.